File: apps/aura-brain/routes/health.js
/**
 * @module routes/health
 * GET /health — Brain service health check.
 */

'use strict';

const express = require('express');
const router = express.Router();
const { createHealthResponse } = require('@aura/shared-types');

const startTime = Date.now();

router.get('/', (req, res) => {
  const uptimeSeconds = Math.floor((Date.now() - startTime) / 1000);

  const health = createHealthResponse({
    service: 'brain',
    status: 'healthy',
    uptime_seconds: uptimeSeconds,
    version: '1.0.0'
  });

  // Add Brain-specific info
  health.plugins_loaded = req.app.locals.pluginRegistry?.count || 0;
  health.active_tools = req.app.locals.mcpRegistry?.getActiveTools().length || 0;
  health.services_tracked = req.app.locals.serviceRegistry?.getServiceNames().length || 0;
  health.service_registry = req.app.locals.serviceRegistry?.getRegistryState() || {};

  res.json(health);
});

module.exports = router;

File: apps/aura-hermes/routes/health.py
"""
Health route — GET /health.
"""

import time
from fastapi import APIRouter

router = APIRouter()

START_TIME = time.time()


@router.get("/health")
async def health():
    """Service health check."""
    uptime_seconds = int(time.time() - START_TIME)
    return {
        "service": "hermes",
        "status": "healthy",
        "uptime_seconds": uptime_seconds,
        "version": "1.0.0",
        "agents_count": 0
    }
